Sources/StreamChat/WebSocketClient/Events/MemberEvents.swift

```diff
@@ -4,41 +4,112 @@
 
 import Foundation
 
-public struct MemberAddedEvent: MemberEvent, ChannelSpecificEvent, EventWithPayload {
-    public let memberUserId: UserId
+public struct MemberAddedEvent: MemberEvent, HasCid, HasUser, HasMember {
     public let cid: ChannelId
+    public let user: ChatUser
+    public let member: ChatChannelMember
+    public let createdAt: Date
+}
+
+extension MemberAddedEvent {
+    struct DTO: EventWithPayload {
+        let cid: ChannelId
+        let user: UserPayload
+        let member: MemberPayload
+        let createdAt: Date
+        let payload: Any
+        
+        init(from response: EventPayload) throws {
+            cid = try response.value(at: \.cid)
+            user = try response.value(at: \.user)
+            member = try response.value(at: \.memberContainer?.member)
+            createdAt = try response.value(at: \.createdAt)
+            payload = response
+        }
+    }
     
-    let payload: Any
-    
-    init(from response: EventPayload) throws {
-        memberUserId = try response.value(at: \.memberContainer?.member?.user.id)
-        cid = try response.value(at: \.cid)
-        payload = response
+    init?(event: DTO, session: DatabaseSession) {
+        guard
+            let user = session.user(id: event.user.id),
+            let member = session.member(userId: event.member.user.id, cid: event.cid)
+        else { return nil }
+        
+        self.init(
+            cid: event.cid,
+            user: user.asModel(),
+            member: member.asModel(),
+            createdAt: event.createdAt
+        )
     }
 }
 
-public struct MemberUpdatedEvent: MemberEvent, ChannelSpecificEvent, EventWithPayload {
-    public let memberUserId: UserId
+public struct MemberUpdatedEvent: MemberEvent, HasCid, HasUser, HasMember {
     public let cid: ChannelId
+    public let user: ChatUser
+    public let member: ChatChannelMember
+    public let createdAt: Date
+}
+
+extension MemberUpdatedEvent {
+    struct DTO: EventWithPayload {
+        let cid: ChannelId
+        let user: UserPayload
+        let member: MemberPayload
+        let createdAt: Date
+        let payload: Any
+        
+        init(from response: EventPayload) throws {
+            cid = try response.value(at: \.cid)
+            user = try response.value(at: \.user)
+            member = try response.value(at: \.memberContainer?.member)
+            createdAt = try response.value(at: \.createdAt)
+            payload = response
+        }
+    }
     
-    let payload: Any
-    
-    init(from response: EventPayload) throws {
-        memberUserId = try response.value(at: \.memberContainer?.member?.user.id)
-        cid = try response.value(at: \.cid)
-        payload = response
+    init?(event: DTO, session: DatabaseSession) {
```

Comment:
```suggestion
    init?(eventDTO: DTO, session: DatabaseSession) {
```

Reply:
Made it a little bit differently: an `EventDTO` type as `toDomainEvent` func so the naming no longer makes sense since the DTO is accessed as `self`.